ddtrace/ext/git.py

```diff
@@ -87,48 +87,64 @@ def extract_user_info(cwd=None):
     # type: (Optional[str]) -> Dict[str, Tuple[str, str, str]]
     """Extract commit author info from the git repository in the current directory or one specified by ``cwd``."""
     # Note: `git show -s --format... --date...` is supported since git 2.1.4 onwards
-    stdout = _git_subprocess_cmd("show -s --format=%an,%ae,%ad,%cn,%ce,%cd --date=format:%Y-%m-%dT%H:%M:%S%z", cwd=cwd)
+    stdout = git_subprocess_cmd("show -s --format=%an,%ae,%ad,%cn,%ce,%cd --date=format:%Y-%m-%dT%H:%M:%S%z", cwd=cwd)
     author_name, author_email, author_date, committer_name, committer_email, committer_date = stdout.split(",")
     return {
         "author": (author_name, author_email, author_date),
         "committer": (committer_name, committer_email, committer_date),
     }
 
 
+def extract_remote_url(cwd=None):
+    remote_url = git_subprocess_cmd("config --get remote.origin.url", cwd=cwd)
+    return remote_url
+
+
+def extract_latest_commits(cwd=None):
+    latest_commits = git_subprocess_cmd("log --format=%H -n 1000", cwd=cwd)
+    return latest_commits.split("\n") if latest_commits else []
+
+
+def get_rev_list_excluding_commits(commit_shas, cwd=None):
+    exclusions = " ".join(["^%s" % sha for sha in commit_shas])
+    commits = git_subprocess_cmd("rev-list --objects --filter=blob:none HEAD %s" % exclusions, cwd=cwd)
```

Comment:
In addition to Yun's comment, we need to pass `--no-object-names`. JS implementation for reference: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/util/git.js#L55 
```suggestion
    commits = git_subprocess_cmd("rev-list --objects --no-object-names --filter=blob:none --since="1 month ago" HEAD %s" % exclusions, cwd=cwd)
```